internal/lang/eval.go

```diff
@@ -289,21 +289,32 @@ func (s *Scope) EvalReference(ref *addrs.Reference, wantType cty.Type) (cty.Valu
 // this type offers, but this is here for less common situations where the
 // caller will handle the evaluation calls itself.
 func (s *Scope) EvalContext(refs []*addrs.Reference) (*hcl.EvalContext, tfdiags.Diagnostics) {
-	return s.evalContext(refs, s.SelfAddr)
+	return s.evalContext(nil, refs, s.SelfAddr)
 }
 
-func (s *Scope) evalContext(refs []*addrs.Reference, selfAddr addrs.Referenceable) (*hcl.EvalContext, tfdiags.Diagnostics) {
+// EvalContextWithParent is exactly the same as EvalContext except the resulting hcl.EvalContext
+// will be derived from the given parental hcl.EvalContext.
```

Comment:
nit: I wonder if we should have a few more words here explaining what this actually means and does in effect.
Even `EvalContext`'s comments on `parent` and `NewChild` don't really explain what it means to use them

Reply:
I extended the comment to describe a little more and also named a few examples: [c3e8fda](https://github.com/opentofu/opentofu/pull/1728/commits/c3e8fda7caf88a09f6e832ab705740c8da33faf6).